packages/big-design/src/components/Tree/types.ts

```diff
@@ -0,0 +1,42 @@
+export interface TreeProps<T> {
+  iconless?: boolean;
+  nodes: TreeNodeProps<T>[];
+  selectable?: 'radio' | 'multi';
+  onExpand?(node: TreeNodeProps<T>): TreeNodeRef<T> | void;
+  onCollapse?(node: TreeNodeProps<T>): TreeNodeRef<T> | void;
+  // Based on the selectable prop, it can either
```

Comment:
Instaed of this comment, lets make the types act that way. :pray:

Reply:
So after attempting two different methods, this isn't possible without casting types.

Both of the attempts used different strategies; one was blocked by not being able to use `undefined` as an interfaces index type, the other was blocked by the idea of variance in typings ([covariance and contravariance](https://www.stephanboyer.com/post/132/what-are-covariance-and-contravariance)).

Going to leave it as is for now.